central/processlisteningonport/store/postgres/full_store.go

```diff
@@ -38,6 +38,9 @@ const getByDeploymentStmt = "SELECT plop.id, plop.serialized, " +
 	"ON plop.processindicatorid = proc.id " +
 	"WHERE plop.deploymentid = $1 AND plop.closed = false"
 
+//const getPodsStmt = "SELECT id from pods"
+const getPodsStmt = "SELECT name from pods where deploymentid = $1"
```

Comment:
Unfortunately it is not possible to match on the pod UID as the listening endpoint does not have that information. So instead match on the pod name. This could be problematic as a pod could be deleted and the pod that replaces it could have the same name. This is unlikely in a deployment, but if it is just a plain pod, it could have the same name. I don't know if it is possible to match on the container.

Reply:
Try to use just one query by matching on PodUid.